pkg/ccl/changefeedccl/changefeed.go

```diff
@@ -37,33 +43,51 @@ type ChangefeedConfig struct {
 
 // makeChangefeedConfigFromJobDetails creates a ChangefeedConfig struct from any
 // version of the ChangefeedDetails protobuf.
-func makeChangefeedConfigFromJobDetails(d jobspb.ChangefeedDetails) ChangefeedConfig {
+func makeChangefeedConfigFromJobDetails(
+	d jobspb.ChangefeedDetails, ctx context.Context, execCfg *sql.ExecutorConfig,
+) (ChangefeedConfig, error) {
+	targets, err := AllTargets(d, ctx, execCfg)
+	if err != nil {
+		return ChangefeedConfig{}, err
+	}
 	return ChangefeedConfig{
 		SinkURI:  d.SinkURI,
 		Opts:     changefeedbase.MakeStatementOptions(d.Opts),
 		ScanTime: d.StatementTime,
 		EndTime:  d.EndTime,
-		Targets:  AllTargets(d),
-	}
+		Targets:  targets,
+	}, nil
 }
 
 // AllTargets gets all the targets listed in a ChangefeedDetails,
 // from the statement time name map in old protos
 // or the TargetSpecifications in new ones.
-func AllTargets(cd jobspb.ChangefeedDetails) (targets changefeedbase.Targets) {
+func AllTargets(
+	cd jobspb.ChangefeedDetails, ctx context.Context, execCfg *sql.ExecutorConfig,
+) (targets changefeedbase.Targets, err error) {
 	// TODO: Use a version gate for this once we have CDC version gates
 	if len(cd.TargetSpecifications) > 0 {
 		for _, ts := range cd.TargetSpecifications {
 			if ts.DescID > 0 {
-				if ts.StatementTimeName == "" {
-					ts.StatementTimeName = cd.Tables[ts.DescID].StatementTimeName
+				if ts.Type == jobspb.ChangefeedTargetSpecification_DATABASE {
```

Comment:
nit: would prefer to see a switch statement instead with a default case returning an assertion error for unhandled types